### PR TITLE
Completion support for kafka file

### LINKED ISSUE #146: Completion support for properties kafka file
It should be nice to support completion inside kafka file:

 * completion for properties for CONSUMER and PRODUCER (ex: topic, from, etc for CONSUMER)
 * completion for key-format, value-format value (with string, long (de)serializers

### PR DESCRIPTION
**Pull request by angelozerr**
Completion support for kafka file

Fixes #146

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
Completion for static properties names and values should work. Test exists but I need to update doc and clean my code.

**Comment by fbricon** [comment]
I get no completion on CONSUMER / PRODUCERR

**Comment by fbricon** [comment]
after completing something, you're still in. placeholder mode, you don't get the next completion unless you press esc or tab
Apr-06-2021 11-08-35

File: src/kafka-file/languageservice/model.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
+export interface ModelDefinition {
+    name: string;
+    description: string;
+    enum?: ModelDefinition[];
+}
+
```
Comment (fbricon) [inline src/kafka-file/languageservice/model.ts:6]: would be preferable if the documentation linked to the embedded doc instead

Comment (angelozerr) [reply src/kafka-file/languageservice/model.ts:6]: I tried, but it seems that it's not supported.

**Comment by angelozerr** [comment]
> I get no completion on CONSUMER / PRODUCER

Could you please give me more information about this problem?

**Comment by fbricon** [comment]
Is there a way for the static snippets to not be available when editing in the middle of a block? i.e avoid:
Screenshot 2021-04-06 at 14 07 05

**Comment by angelozerr** [comment]
> after completing something, you're still in. placeholder mode, you don't get the next completion unless you press esc or tab

It's the snippet vscode problem. I fear that there is no fix for that.

**Comment by fbricon** [comment]
if you type PR or CO, you don't get completion for just the keyword PRODUCER or CONSUMER

**Comment by angelozerr** [comment]
> Is there a way for the static snippets to not be available when editing in the middle of a block? i.e avoid:

The only solution is to move those snippet in the language service. I don't know if it's easy to do that. Please create a new issue about that.

**Comment by angelozerr** [comment]
> if you type PR or CO, you don't get completion for just the keyword PRODUCER or CONSUMER

You speak about snippet completion (managed by vscode)?

**Comment by fbricon** [comment]
please update the changelog

**Comment by angelozerr** [comment]
> please update the changelog

fixed

**Pull request merged**